src/dbnode/storage/series/series.go

```diff
@@ -414,6 +415,19 @@ func (s *dbSeries) FetchIndexChecksum(
 	return r, err
 }
 
+func (s *dbSeries) FetchReadMismatches(
+	ctx context.Context,
+	batchReader wide.IndexChecksumBlockBatchReader,
+	blockStart time.Time,
+	nsCtx namespace.Context,
+) (wide.StreamedMismatchBatch, error) {
+	s.RLock()
+	reader := NewReaderUsingRetriever(s.id, s.blockRetriever, s.onRetrieveBlock, s, s.opts)
+	r, err := reader.FetchReadMismatches(ctx, batchReader, blockStart, nsCtx)
+	s.RUnlock()
```

Comment:
Do we need to hold the lock for the duration of the read?

Reply:
Yes, we are writing into the series.